inst/tutorials/tidy-01-causality1/tidy-01-causality1.Rmd

```diff
@@ -0,0 +1,377 @@
+  pass_if(~ identical(.result, 0.25))
+)
+```
+
+
+### Comparing logicals
+
+
+We often combine logical statements using AND (`&`) and OR (`|`) in R. For AND statements, both expressions have to be true for the whole expression to be true:
+
+- `TRUE & FALSE`, `FALSE & TRUE`, and `FALSE & FALSE` are `FALSE`
+- `TRUE & TRUE` is `TRUE`
+
+For OR statements, either statement being true makes the whole expression true:
+
+- `TRUE | FALSE`, `FALSE | TRUE`, and `TRUE | TRUE` are `TRUE`
+- `FALSE | FALSE` is `FALSE`
+
+
+```{r comparing-logicals, echo=FALSE}
+question("What does expression `(TRUE | FALSE) & TRUE` evaluate to?",
+  answer("`TRUE`", correct = TRUE),
+  answer("`FALSE`"),
+  answer("`NA`")
+)
+```
+
+
+### Comparing objects
+
+
+There are several *relational operators* that allow us to compare objects in R. The most useful of these are the following:
+
+- `>` greater than, `>=` greater than or equal to
+- `<` less than, `<=` less than or equal to
+- `==` equal to
+- `!=` not equal to
+
+When we use these to compare two objects in R, we end us with a logical object. You can also compare a vector to a particular number.
+
+
+#### Exercises
+
+- Test if 10 is greater than 5.
+
+```{r gt-test, exercise=TRUE}
+
+```
+
+```{r gt-test-solution}
+10 > 5
+```
+
+```{r gt-test-check}
+grade_code()
+```
+
+- Test which values in the vector `x` is greater than or equal to 0.
+
+```{r compare, exercise = TRUE}
+## x vector
+x <- c(-2, -1, 0, 1, 2)
+
+## test which values of x are greater than or equal to 0
+
+```
+
+```{r compare-check}
+grade_result(
+  fail_if(~ identical(.result, x > 0),
+          "Did you forget the 'or equal to' part of the comparison?"),
+  pass_if(~ identical(.result, x >= 0))
+)
+```
+
+## Subsets in R
+
+### Subsetting a data frame
+
+
+You can use the same logical statements you have been using to create subsets of a data frame. These can often be helpful because we'll want to calculate various quantities of interest for different subsets of the data. For this exercise, we will use the `resume` data frame made up of the variables `firstname`, `sex`, `race`, and `call`. As a reminder, here is what the data look like:
+
+```{r}
+resume
+```
+
+#### Exercise
+
+- Use the `filter` function to create a subset of the `resume` data frame that is only female names that sound white. Save this subset as `resume.wf`
+- Use the `head` function to see the first six lines of this subset.
+- Use the `summarize` function to calculate the mean of the `call` variable in this subset. Save this output as `resume.wf.callback`.
+
+
+```{r subset, exercise = TRUE}
+## create the subset for white female names and
+## assign it to resume.wf
+resume.wf <- ...
+
+## print the first 6 lines of the subset
+
+
+## calculate the mean of the callback variable (call)
+
+```
+
+```{r subset-solution}
+resume.wf <- resume %>%
+  filter(race == "white" & sex == "female")
+
+head(resume.wf)
+
+resume.wf.callback <- resume.wf %>%
+  summarize(mean(call))
```

Comment:
Not sure if there's anything to do about it, but this fails if the give the mean a name like `summarize(callmean = mean(call))`.

Reply:
This is a practical trick so might be better for students to pick it up. Added a short follow up question about naming variables in summarize() directly but we can also do something different if this feels unnecessary.